Python/Product/Analysis/Interpreter/Ast/AstPythonFunction.cs

```diff
@@ -40,31 +45,25 @@ class AstPythonFunction : IPythonFunction, ILocatedMember {
                 }
             }
 
-            _overloads = new List<AstPythonFunctionOverload> {
-                new AstPythonFunctionOverload(Documentation, "", MakeParameters(ast, def), MakeReturns(def))
-            };
+            _overloads = new List<IPythonFunctionOverload>();
 
             Locations = new[] { loc };
         }
 
-        internal void AddOverload(PythonAst ast, FunctionDefinition def) {
-            _overloads.Add(new AstPythonFunctionOverload(def.Documentation, "", MakeParameters(ast, def), MakeReturns(def)));
+        internal void AddOverload(IPythonFunctionOverload overload) {
+            _overloads.Add(overload);
         }
 
-        private static IEnumerable<IParameterInfo> MakeParameters(PythonAst ast, FunctionDefinition def) {
+        internal static IEnumerable<IParameterInfo> MakeParameters(PythonAst ast, FunctionDefinition def) {
             foreach (var p in def.Parameters) {
                 yield return new AstPythonParameterInfo(ast, p);
             }
         }
 
-        private static IEnumerable<IPythonType> MakeReturns(FunctionDefinition def) {
-            yield break;
-        }
-
         public IPythonModule DeclaringModule {get;}
         public IPythonType DeclaringType {get;}
         public string Name { get; }
-        public string Documentation {get;}
+        public string Documentation => _overloads.FirstOrDefault()?.Documentation;
```

Comment:
I wonder if this should rather be `_overloads.Select(o => o.Documentation).Where(doc => doc != null).FirstOrDefault()`, so as to get the first available docstring, rather than the first available overload.